rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -342,6 +342,20 @@ class Client : public ClientBase
   std::shared_future<typename GoalHandle::SharedPtr>
   async_send_goal(const Goal & goal, const SendGoalOptions & options = SendGoalOptions())
   {
+    // To prevent the list from growing out of control, forget about any goals
```

Comment:
I'm not a fan of cleaning up goal handles in the send goal function, but I wasn't sure of a better place to put this logic (or an alternative to ensure we don't grow the map of goal handles indefinitely).